### ISSUE TITLE
How does this work?

### ISSUE THREAD

**Issue by deprale** [luketimothyjones/YouTube-Spam-Remover#1]
So this basically removes any comment that has either
1) an author that has emojis in his name (a lot of legitimate users do that)
2) somebody linked something? (a lot of legitimate links)

or maybe i'm stupid and I don't get the big idea, I don't see anything that will rate a comment based on some spam score using some advanced algorithm.

I mean if it does that, then it is pretty useless(no offense)

I see why it would work mainly on these porn comment bots, and those annoying bots that post "Finally it's arrived 'link-here'" but I don't think it's a good solution, it will block most bots though but it will also block a lot of legitimate users.

Pardon me for any misunderstanding, I'm not a javascript expert.

EDIT: What you COULD do instead, maybe like scrape for a bit a lot of comments and comment replies, and sort them out by most repeating, you will find lots of spam! Now if you already have a paid-for VPS just host a mysql server, and implement a button to report for spam in this extension, and additionally block anything that's found in the database (I'm unsure how you would tackle such an issue, maybe you could cache the database and let the extension periodically update from it as to avoid congestion and performance issues) and then when users report same pattern more than n times then add it to the blocklist, so you sort of add like a uhh... universal spam block thingy based on real users, from then on it's a self-sustaining project and you might get a bit of donations out of your work! Just an idea though... I'm no expert.

**Comment by samipfjo** [comment]
Hi deprale,

So you are correct on several counts. There is no advanced spam-detection algorithm, the filtering is based upon a couple of observations I've made regarding the content of spam comments.

Addressing your first point, the fastest blocking method implemented is indeed a ban on emojis in usernames, as up until recently I had not seen legitimate usernames that included them. Modifying the regex to only block emojis that are commonly used by spammers (hands pointing to username, peaches/eggplant, etc.) is on my to-do list.

Addressing your second point, you're somewhat correct, but only _uncommon_ links are blocked. I have constructed a list of 300,000 permissible websites by taking a custom list of the most frequently visited websites from [the Tranco project](https://tranco-list.eu/) and removing domains that are in a [collection of spam/malware/abuse](https://github.com/blocklistproject/Lists) from it. YouTube links are not blocked, as filtering spam from those would be a bit of a nightmare and super prone to false positives.

Lastly, maintaining a database of spam comments is something that I've considered, and while it would be really cool I unfortunately cannot afford the hosting costs for a VPS.

Thanks for your interest in the project and your questions!

Edit:
Additionally, pinned comments and comments by the video poster are ignored, so they can comment with whatever links they like.
